app/assets/javascripts/models/page/search_page.js.coffee

```diff
@@ -50,7 +50,7 @@ ns.SearchPage = do (ko
     $('.launch-customize-modal').click ->
       $('#customizeDataModal').modal('show')
 
-    setTimeout initModal, 2000 if !window.edscportal
+    preferences.ready.done(-> initModal()) if !window.edscportal
 
   class SearchPage
     constructor: ->
```

Comment:
@mightynimble Also, to make it clear - the race condition is due to the usage of an existing AJAX call in jQuery.  One could argue that we are using a jQuery based solution to fix a jQuery based problem.